Source/Snowman.cpp

```diff
@@ -188,13 +203,15 @@ void Snowman::translateSnowman(GLFWwindow* window, const bool& shift, bool& canM
 
 	// Snowman Translation by WASD keys
 	float snowmanTranslationSpeed = globalSpeed * 0.03f;
-	bool moveLeft = glfwGetKey(window, GLFW_KEY_A) == GLFW_PRESS;
+	bool moveLeft  = glfwGetKey(window, GLFW_KEY_A) == GLFW_PRESS;
 	bool moveRight = glfwGetKey(window, GLFW_KEY_D) == GLFW_PRESS;
-	bool moveUp = glfwGetKey(window, GLFW_KEY_W) == GLFW_PRESS;
-	bool moveDown = glfwGetKey(window, GLFW_KEY_S) == GLFW_PRESS;
+	bool moveUp    = glfwGetKey(window, GLFW_KEY_W) == GLFW_PRESS;
+	bool moveDown  = glfwGetKey(window, GLFW_KEY_S) == GLFW_PRESS;
+
+	adjustKeysToRotation();
 
 	// Move snowman left (+ x direction)
-	if (moveLeft) {
+	if (moveLeft && isFreeRight) {
```

Comment:
Similar to my last comment, 
Should be moveLeft && isFreeLEFT. Compare to check below for moveUp && isFreeForwards. Also fishy.